app/src/main/java/org/keepgoeat/presentation/home/HomeMyGoalAdapter.kt

```diff
@@ -0,0 +1,109 @@
+    ) : RecyclerView.ViewHolder(binding.root) {
+        var layout = binding
+        fun bind(myGoal: MyGoalInfo, goalType: HomeGoalType) {
+            val btnType: HomeBtnType = if (goalType == HomeGoalType.MORE) { // 더 먹기인 경우
+                if (myGoal.goalAchieved) {
+                    HomeBtnType.PLUS_ACHIEVED
+                } else {
+                    HomeBtnType.PLUS_NOT_ACHIEVED
+                }
+            } else { // 덜 먹기인 경우
+                if (myGoal.goalAchieved) {
+                    HomeBtnType.MINUS_ACHIEVED
+                } else {
+                    HomeBtnType.MINUS_NOT_ACHIEVED
+                }
+            }
+            binding.goal = myGoal
+            binding.goalType = goalType
+            binding.goalBtn = btnType
+        }
+    }
+
+    class AddGoalViewHolder(
+        private val binding: ItemAddGoalBinding
+    ) : RecyclerView.ViewHolder(binding.root) {
+        var layout = binding
+        fun bind(goalCount: Int) {
+            when (goalCount) {
+                0 -> {}
+                1 -> binding.tvAddMoreGoal.setText(R.string.home_two_more_goal)
+                2 -> binding.tvAddMoreGoal.setText(R.string.home_one_more_goal)
+                3 -> {
+                    binding.tvAddMoreGoal.setText(R.string.home_no_more_goal)
+                    binding.btnMakeGoal.setVisibility(false)
+                }
+            }
+        }
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        return when (viewType) {
+            MyGoalInfo.MY_GOAL_TYPE -> {
+                val binding =
+                    ItemMyGoalBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+                MyGoalViewHolder(binding)
+            }
+            MyGoalInfo.ADD_GOAL_TYPE -> {
+                val binding = ItemAddGoalBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+                AddGoalViewHolder(binding)
+            }
+            else -> {
+                throw java.lang.ClassCastException("Unknown ViewType Error")
+            }
+        }
+    }
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        when (currentList[position].type) {
+            MyGoalInfo.MY_GOAL_TYPE -> {
+                if (currentList[position].moreGoal) {
+                    (holder as MyGoalViewHolder).bind(currentList[position], HomeGoalType.MORE)
+                } else {
+                    (holder as MyGoalViewHolder).bind(currentList[position], HomeGoalType.LESS)
+                }
+                holder.layout.btnGoal.setOnClickListener {
+                    currentList[position].goalAchieved = !currentList[position].goalAchieved
+                    notifyItemChanged(position)
+                }
+            }
+            MyGoalInfo.ADD_GOAL_TYPE -> {
+                (holder as AddGoalViewHolder).bind(currentList.size - 1)
+            }
```

Comment:
```suggestion
            when (holder) {
                is MyGoalViewHolder -> { 
                    holder.bind(currentList[position], if(currentList[position].moreGoal) HomeGoalType.MORE else HomeGoalType.LESS)
                    holder.layout.btnGoal.setOnClickListener {
                        currentList[position].goalAchieved = !currentList[position].goalAchieved
                        notifyItemChanged(position)
                    }
                }
                is AddGoalViewHolder -> { holder.bind(currentList.size - 1) }
            }
```

이렇게 수정해보시면 더 깔끔할 것 같습니다~ as로 타입 캐스팅도 필요 없구용!!
근데 클릭 리스너를 뷰홀더 내가 아닌 onBindViewHolder()에 넣으신 이유가 있을까용?!

Reply:
그 currentList에서 목표 달성여부 boolean값 바꾸고 어댑터에 변경사항을 알려야 되는데 뷰홀더 내에서는 참조가 안되고 notifyItemChanged같은 함수 호출이 안되는거 같아서 onBindViewHolder에 넣었어용
혹시 뷰홀더에서 적용하는 방법이 있을까용?

Reply:
해결 완료!